### PR TITLE
Listing conversations

### PR DESCRIPTION
**Pull request by albertsola**

### CODE REVIEW

File: .vscode/settings.json
Comment (albertsola) [inline .vscode/settings.json:1]: can you add your .vscode into your local global ignore

https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files

File: rxchat/server/chat_channel.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from typing import List
+
+from pydantic import BaseModel
+
+class ChatChannel(BaseModel):
+    id: str
+    users_count: int
```
Comment (albertsola) [inline rxchat/server/chat_channel.py]: I would probably return the list of usernames connected to the specific channel. Then the frontend can count them if required, or list them.

File: rxchat/server/chat_channel.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from typing import List
+
+from pydantic import BaseModel
+
+class ChatChannel(BaseModel):
```
Comment (albertsola) [inline rxchat/server/chat_channel.py]: Just for keeping naming uniform I will call it ConversationListItem

File: rxchat/server/api.py [hunk 3]
```diff
@@ -13,3 +15,24 @@ async def connect_chat(websocket: WebSocket):
         await chat_server.handle_user_websocket(username, websocket)
     finally:
         await chat_server.handle_user_disconnected(username)
+
+@router.get("/chat/{conversation_id}")
+async def get_conversation_id(conversation_id: str) -> Conversation:
```
Comment (albertsola) [inline rxchat/server/api.py]: rename chat to conversation

File: rxchat/server/api.py [hunk 4]
```diff
@@ -13,3 +15,24 @@ async def connect_chat(websocket: WebSocket):
...
+        if not channel_conversation:
+            channel_users_count = 0
+        else:
+            channel_users_count = channel_conversation.user_count()
+
+        response.append({
+            "id": channel_id,
+            "users_count": channel_users_count
+        })
+    return response
```
Comment (albertsola) [inline rxchat/server/api.py]: rename channel(s) to conversation(s) for now

File: rxchat/component/state.py [hunk 5]
```diff
@@ -1,36 +1,41 @@
...
+from rxchat.server.chat_channel import ChatChannel
 
 class ChatState(rx.State):
     """The app state."""
 
     _chat: ChatClient | None = None
     connected: bool = False
-    channels: list[str] = ["Welcome", "Jokes", "Tech"]
+    channels_data: dict[str, ChatChannel] = {}
+    channels: list[str] = ["Welcome"]
```
Comment (albertsola) [inline rxchat/component/state.py]: I would call for now conversations instead of channels.

**Review by albertsola — COMMENTED** [review]
Great job!

I have a few minor notes around the naming.

While running the app I got the following warning which we can address later:

UserWarning: Mixing V1 models and V2 models (or constructs, like `TypeAdapter`) is not supported. Please upgrade `Conversation` to V2.

File: rxchat/server/chat_channel.py (hunk 1 above)
Comment (starx) [reply rxchat/server/chat_channel.py]: Fair enough.

File: rxchat/server/api.py (hunk 3 above)
Comment (starx) [reply rxchat/server/api.py]: ok

**Pull request merged**